contracts/utils/Arrays.sol

```diff
@@ -23,55 +23,151 @@ library Arrays {
      * when executing this as part of a transaction. If the array being sorted is too large, the sort operation may
      * consume more gas than is available in a block, leading to potential DoS.
      */
+    function sort(
+        bytes32[] memory array,
+        function(bytes32, bytes32) pure returns (bool) comp
+    ) internal pure returns (bytes32[] memory) {
+        uint256 begin;
+        uint256 end;
+        /// @solidity memory-safe-assembly
+        assembly {
+            begin := add(array, 0x20)
+            end := add(begin, mul(mload(array), 0x20))
+        }
```

Comment:
What about having these functions in Array.sol:

```solidity
function begin(bytes32[] memory array) internal pure returns (bytes32 ptr) {
  assembly { ptr := add(array, 0x20) }
} 

function end(bytes32[] memory array) internal pure returns (bytes32 ptr) {
  assembly { ptr := add(begin, mul(mload(array), 0x20)) }
} 
```

So we can do:

```solidity
_quickSort(array.start(), array.end());
```

Reply:
done.